share/availability/discovery/discovery.go

```diff
@@ -57,6 +61,19 @@ func NewDiscovery(
 		peersLimit,
 		discInterval,
 		advertiseInterval,
+		nil,
+	}
+}
+
+func (d *Discovery) WithOnPeersUpdate(f OnPeersUpdate) {
```

Comment:
I would make it via a channel to protect `OnPeersUpdate` from blocking `handlePeerFound` (API hygiene), but this is also fine

Reply:
Some basic test would hurt